File: collaboration-client/NimbleCollaborationClient/Type/ReadResourceType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Web.Script.Serialization;

namespace Nimble.Client.Type
{
    public class ReadResourceType: JsonType
    {

        public ReadResourceType() { }
    
        public ReadResourceType(String token, String projectID, String name):this(token, projectID, name, 0) { 
    	    
        }

        public ReadResourceType(String token, String projectName, String name, int? ver)
        { 
    	    this.token = token;
    	    this.projectName = projectName;
    	    this.resourceName = name;
    	    this.resourceVersion = ver;
        }

        public String token {get; set;}
        public String projectName { get; set; }
        public String resourceName { get; set; }
        public int? resourceVersion { get; set; }

        public static ReadResourceType mapJson(String json) {
		    try {
                return new JavaScriptSerializer().Deserialize<ReadResourceType>(json);
		    } catch (Exception e) {
			    return null;
		    }
	    }


    }
}

File: collaboration-service/src/main/java/it/domina/nimble/collaboration/services/type/ReadResourceType.java
package it.domina.nimble.collaboration.services.type;

import org.apache.log4j.Level;
import org.apache.log4j.Logger;

import com.fasterxml.jackson.databind.ObjectMapper;

import it.domina.nimble.utils.JsonType;

public class ReadResourceType extends JsonType {

    private static final Logger logger = Logger.getLogger(ReadResourceType.class);

   	private String token;
   	private String projectName;
   	private String resourceName;
   	private Integer resourceVersion;
	
    public ReadResourceType() { }
    
    public ReadResourceType(String token, String projectID, String name) { 
    	this(token, projectID, name, null);
    }

    public ReadResourceType(String token, String projectName, String name, Integer ver) { 
    	this.token = token;
    	this.projectName = projectName;
    	this.resourceName = name;
    	this.resourceVersion = ver;
    }

    public String getToken() {
    	return this.token;
    }

    public String getProjectName() {
    	return this.projectName;
    }

    public String getResourceName(){
    	return this.resourceName;
    }

    public Integer getResourceVersion(){
    	return this.resourceVersion;
    }
    
    public static ReadResourceType mapJson(String json) {
		try {
        	ObjectMapper mapper = new ObjectMapper();
			return mapper.readValue(json, ReadResourceType.class);
		} catch (Exception e) {
			logger.log(Level.INFO, e);
			return null;
		}
	}

	
}
